tpp-app/src/services/utils/helpers.js

```diff
@@ -16,3 +18,29 @@ const initializeEmptyYear = (yearNumber) => {
 
   return year
 }
+
+
+/**
+ * Convert a Date object into a date string, encoding year, month and day. Note it encodes months as 1 indexed, and days as 0 indexed
+ * @param {Date} date Object to convert to string
+ * @return {String} String encoding year, month and day
+ */
+export const getDateString = (date) => {
+  var date = date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + date.getDate();
+  return date;
+
+}
+
+/**
+ * Retrieves the user's symptom data for the given date.
+ * @param day number
+ * @param month number (January = 0)
+ * @param year number
+ */
+ export const GETsymptomsForDate = async (day, month, year) => {
+  // Get the year's data (value could be null if year is empty)
+  const yearData = JSON.parse(await AsyncStorage.getItem(year.toString()));
+
+  // Return symptoms for that day or empty symptoms object if it doesn't exist
+  return yearData[month-1][day-1] ? new Symptoms(yearData[month-1][day-1]) : new Symptoms();
```

Comment:
Loving this! so if something goes wrong, return Symptoms obj with everything as null. No unexpected surprises. 

One Q: if the passed in day month and year are invalid what will happen? Just in case we should guard against an IndexOutOfBounds when accessing ```yearData[month-1][day-1]```

Reply:
I'll talk it over with Emily but I would think returning `Symptoms` with everything null works? I think that we can usually treat the UNRECORDED state and the INVALID state for a day's symptoms the same way